packages/core-flows/src/definition/cart/steps/find-sales-channel.ts

```diff
@@ -0,0 +1,51 @@
+import { ModuleRegistrationName } from "@medusajs/modules-sdk"
+import { ISalesChannelModuleService, SalesChannelDTO } from "@medusajs/types"
+import { MedusaError } from "@medusajs/utils"
+import { StepResponse, createStep } from "@medusajs/workflows-sdk"
+
+interface StepInput {
+  salesChannelId?: string
+  publishableApiKeyScopes?: {
+    salesChannelIds?: string[]
+  }
+}
+
+export const findSalesChannelStepId = "find-sales-channel"
+export const findSalesChannelStep = createStep(
+  findSalesChannelStepId,
+  async (data: StepInput, { container }) => {
+    const salesChannelService = container.resolve<ISalesChannelModuleService>(
+      ModuleRegistrationName.SALES_CHANNEL
+    )
+
+    let { salesChannelId, publishableApiKeyScopes } = data
+
+    // TODO: Revisit when API key module is implemented
+    if (salesChannelId && publishableApiKeyScopes?.salesChannelIds?.length) {
+      if (publishableApiKeyScopes.salesChannelIds.length > 1) {
```

Comment:
thought: would probably make the workflow independent of the publishable key and just handle this in the API layer instead.

Reply:
Good push. I wanted to wait for the API key module to put thought into this part of the flow, but I'll just remove it for now.